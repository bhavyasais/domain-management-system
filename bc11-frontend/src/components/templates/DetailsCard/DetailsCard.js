import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { CardHeader, Divider, Grid, Typography, useTheme } from '@material-ui/core';
import IconAtom from '../../atoms/Icon/IconAtom';
import IconButtonAtom from '../../atoms/IconButton/IconButtonAtom';
import options from '../../atoms/shared/ListIcons';
import ButtonAtom from '../../atoms/Button/ButtonAtom';
import Dropdown from '../../molecules/Dropdown/Dropdown';

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: '450px',
    height: '726px',
  },
  threat: {
    height: '25px',
    paddingRight: '20px',
    paddingLeft: '30px',
  },
  source: {
    height: '25px',
    paddingLeft: '1px',
  },
  subject: {
    paddingTop: '20px',
    paddingRight: '20px',
    paddingLeft: '30px',
  },
  margin: {
    marginBottom: '5px',
    color: theme?.palette?.darkGrey?.main,
  },
  contentStyle: {
    color: theme?.palette?.black?.main,
  },
  details: {
    paddingTop: '20px',
    paddingLeft: '30px',
  },
  arrowIcon: {
    transform: 'translate(5px, 5px)',
  },
  dropDown: {
    transform: 'translate(-40px, 15px)',
  },
}));

const DetailsCard = ({ cardDetails, sourceDetailsHeader, threatDetailsHeader, selectedItem }) => {
  const [open, setOpen] = useState(true);

  const showSource = () => {
    setOpen(false);
  };
  const showThreat = () => {
    setOpen(true);
  };
  const classes = useStyles();
  const theme = useTheme();
  return (
    <Card
      className={classes.root}
      variant="outlined"
      style={{ borderColor: theme?.palette?.grey?.main }}
    >
      {!open && (
        <CardHeader
          title={sourceDetailsHeader}
          avatar={
            <IconButtonAtom onClick={showThreat} className={classes.arrowIcon}>
              <IconAtom children={options.arrowBack} />
            </IconButtonAtom>
          }
          action={
            <IconButtonAtom>
              <IconAtom children={options.close} />
            </IconButtonAtom>
          }
          titleTypographyProps={{ variant: 'h4' }}
          className={classes.source}
        />
      )}
      {open && (
        <CardHeader
          title={threatDetailsHeader}
          action={
            <IconButtonAtom>
              <IconAtom children={options.close} />
            </IconButtonAtom>
          }
          titleTypographyProps={{ variant: 'h4' }}
          className={classes.threat}
        />
      )}
      <Divider variant="middle" light />
      <Grid container>
        <Grid container>
          <Grid
            item
            container
            direction="row"
            className={clsx(classes.margin, classes.subject)}
            justify="space-between"
          >
            <Typography variant="h6">{cardDetails.id}: Subject</Typography>
            {open && (
              <ButtonAtom
                textVariant="caption"
                variant="text"
                value="View Source"
                style={{ color: theme?.palette?.green?.main }}
                onClick={showSource}
              />
            )}
          </Grid>
          <Grid container spacing={3} className={classes.details}>
            <Grid item container direction="column">
              <Typography variant="button" className={classes.margin}>
                Domain
              </Typography>
              <Typography variant="body1" className={classes.contentStyle}>
                {cardDetails.Domain}
              </Typography>
            </Grid>

            <Grid item container direction="column">
              <Typography variant="button" className={classes.margin}>
                From
              </Typography>
              <Typography variant="body1" className={classes.contentStyle}>
                {cardDetails.From}
              </Typography>
            </Grid>
            <Grid item container direction="column">
              <Typography variant="button" className={classes.margin}>
                {open ? 'Description' : 'Email'}
              </Typography>
              <Typography
                variant="body1"
                className={classes.contentStyle}
                style={{ width: '378px' }}
              >
                {open ? cardDetails.description : cardDetails.emailDetails}
              </Typography>
            </Grid>
            {open && (
              <>
                <Grid item container direction="column">
                  <Typography variant="button" className={classes.margin}>
                    Treat type
                  </Typography>
                  <Typography variant="body1" className={classes.contentStyle}>
                    {cardDetails.threatType}
                  </Typography>
                </Grid>

                <Grid item container direction="column">
                  <Typography variant="button" className={classes.margin}>
                    Treat level
                  </Typography>
                  <Typography variant="body1" className={classes.contentStyle}>
                    {cardDetails.threatLevel}
                  </Typography>
                </Grid>

                <Grid item container direction="column">
                  <Typography variant="button" className={classes.margin}>
                    Trust score
                  </Typography>
                  <Typography variant="body1" className={classes.contentStyle}>
                    {cardDetails.trustScore}
                  </Typography>
                </Grid>

                <Grid item container direction="column">
                  <Typography variant="button" className={classes.margin}>
                    Source date time
                  </Typography>
                  <Typography variant="body1" className={classes.contentStyle}>
                    {cardDetails.time}
                  </Typography>
                </Grid>
              </>
            )}

            <Grid item container>
              <Typography variant="button" className={classes.margin}>
                Group
              </Typography>
              <Dropdown
                onChange={cardDetails.onChange}
                selectedItem={selectedItem}
                list={cardDetails.trustGroups}
                dropDownStyle={classes.dropDown}
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Card>
  );
};
DetailsCard.propTypes = {
  cardDetails: PropTypes.shape({ value: PropTypes.string, name: PropTypes.string }).isRequired,
};
DetailsCard.defaultProps = {};

export default DetailsCard;
