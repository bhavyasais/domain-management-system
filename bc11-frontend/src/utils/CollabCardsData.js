import box from "../assets/atoms_icons_box.svg";
import gsuite from "../assets/g-suite.jpg";
import okta from "../assets/atoms_icons_okta.svg";
export const collabCardsData1 = [
  {
    title: "box card",
    image: box ,
    component: "img",
    height: "200",
    name: "Box",
  },
  {
    title: "g suite card",
    image:  gsuite ,
    component: "img",
    height: "200",
    name: "G Suite",
  },
  {
    title: "docu sign card",
    image:
      "https://documentautomation.net/wp-content/uploads/2016/05/docusign-circle-large.png",
    component: "img",
    height: "200",
    name: "Docusign",
  },
  {
    title: "figma card",
    image:
      "https://docs.microsoft.com/en-us/windows/uwp/design/downloads/images/figma.png",
    component: "img",
    height: "200",
    name: "Figma",
  },
];
export const collabCardsData2 = [
    {
      title: "okta card",
      image:  okta ,
      component: "img",
      height: "200",
      name: "Okta",
    },
    {
      title: "slack card",
      image: "https://cdn.mos.cms.futurecdn.net/SDDw7CnuoUGax6x9mTo7dd.jpg",
      component: "img",
      height: "200",
      name: "Slack",
    },
  ];
  
