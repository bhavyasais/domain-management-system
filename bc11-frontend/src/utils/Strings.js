export const TRUSTGROUP_COLORS = {
  'Top 100': '#daefa9',
  'Top 10000': '#b0f1f0',
  'Top 30000': '#fdd570',
  'Do not trust': '#ffada5',
};

export const TRUST_GROUPS = ['Top 100', 'Top 1000', 'Top 3000', 'Do not trust'];

export const RELATIONSHIPS = ['Customer', 'Vendor', 'Partner'];

export const RELATIONS = [
  { groupName: 'Customer' },
  { groupName: 'Vendor' },
  { groupName: 'Partner' },
];

export const RELATIONSHIP_COLORS = {
  Customer: '#fdd570',
  Partner: '#b0f1f0',
  Vendor: '#daefa9',
};

export const STATUS = {
  SUCCESS: 'SUCCESS',
  FAILURE: 'FAILURE',
};

export const RELATIONSHIP_CARD_LEGEN_1 = ' Lorem ipsum dolor sit amet, consectetur adipiscing elit';
export const RELATIONSHIP_CARD_LEGEN_2 = ' Lorem ipsum dolor sit amet, consectetur adipiscing elit';
export const DASHBOARD_HEADER_LABEL = 'Dashboard';
export const GREET_USER = 'Good morning,';
export const SUBHEADING = 'Here is whatâ€™s happening with your domains';
export const GRAPH_CARDS = ['Total Domains', 'Total Received', 'Trusted', 'Do Not Trust'];
export const PLACEHOLDER = 'Mail@gmail.com';
export const DESCRIPTION =
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat';
export const SOURCE_DETAILS_HEADER = 'Source Details';
export const THREAT_DETAILS_HEADER = 'Threat Details';
