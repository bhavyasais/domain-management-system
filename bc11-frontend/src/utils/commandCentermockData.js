import React from "react";
import faker from "faker";

const clickedCheckbox = (params) => {};

const checkboxRenderer = (params) => {
  return (
    <div>
      <input type="checkbox" onClick={() => clickedCheckbox(params)} />
    </div>
  );
};

const headerRenderer = (params) => {
  const name = params.displayName;
  return (
    <div>
      <input type="checkbox" onClick={() => clickedCheckbox(params)} />
      <label>{name}</label>
    </div>
  );
};

const redoFunction = (headerName, headerField) => {
  return {
    headerName: headerName,
    field: headerField,
    cellRendererFramework: (params) => {
      return checkboxRenderer(params);
    },
    headerComponentFramework: (params) => {
      return headerRenderer(params);
    },
    cellStyle: { textAlign: "center" },
  };
};

const columnDefs = [
  { headerName: "S no", valueGetter: "node.rowIndex + 1" },
  {
    headerName: "A Trust Group",
    field: "trustGroup",
    sortable: true,
    filter: true,
  },
];

columnDefs.push(redoFunction("Edit File?", "editFile"));
columnDefs.push(redoFunction("View File?", "viewFile"));
columnDefs.push(redoFunction("Delete File?", "deleteFile"));
columnDefs.push(redoFunction("Share File?", "shareFile"));

export const trustGroupcolumnDefs = columnDefs;

export const trustGrouprowData = [
  {
    trustGroup: "DO NOT TRUST",
    editFile: true,
    viewFile: false,
    deleteFile: false,
    shareFile: false,
  },
  {
    trustGroup: "Top 10",
    editFile: false,
    viewFile: false,
    deleteFile: true,
    shareFile: false,
  },
  {
    trustGroup: "Top 10000",
    editFile: false,
    viewFile: false,
    deleteFile: false,
    shareFile: false,
  },
  {
    trustGroup: "Top 30000",
    editFile: false,
    viewFile: false,
    deleteFile: false,
    shareFile: false,
  },
];

export const DomaincolumnDefs = [
  { headerName: "S no", valueGetter: "node.rowIndex + 1" },
  { headerName: "A Domain", field: "domain", sortable: true, filter: true },
  {
    headerName: "Trust Score",
    field: "trustScore",
    sortable: true,
    filter: true,
  },
  {
    headerName: "Trust Group",
    field: "trustGroup",
    sortable: true,
    filter: true,
  },
];

export const DomainrowData = [
  {
    domain: faker.lorem.word(),
    email: "example@yahoo.com",
    trustScore: faker.random.number(),
    trustGroup: "Do not trust",
    id: 0,
  },
  {
    domain: faker.lorem.word(),
    email: "user123@gamil.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Top 100",
    id: 1,
  },
  {
    domain: faker.lorem.word(),
    email: "testEmail@Zemoso.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Do not trust",
    id: 2,
  },
  {
    domain: faker.lorem.word(),
    email: "testUser@Zemoso.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Top 1000",
    id: 3,
  },
  {
    domain: faker.lorem.word(),
    email: "testPerson@Zemoso.com", //NOSONAR
    trustScore: faker.random.number(),
    trustGroup: "Top 3000",
    id: 4,
  },
];
export const RelationshipData = [
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Partner",
  },
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Vendor",
  },
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Customer",
  },
  {
    domain: faker.lorem.word(),
    rustScore: faker.random.number(),
    relationship: "Partner",
  },
  {
    domain: faker.lorem.word(),
    trustScore: faker.random.number(),
    relationship: "Vendor",
  },
];
