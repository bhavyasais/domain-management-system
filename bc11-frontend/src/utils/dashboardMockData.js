import React from 'react';

export const graphCardDetails = [
  {
    cardName: 'Total Domains',
    totalCount: 30,
    additionalCount: 1,
  },
  {
    cardName: 'Total Received',
    totalCount: 30,
    additionalCount: 5,
  },
  {
    cardName: 'Trusted',
    totalCount: 30,
    additionalCount: 5,
  },
  {
    cardName: 'Do Not Trust',
    totalCount: 30,
    additionalCount: 1,
  },
];

export const graphData = {
  value: 'zemoso',
  name: 'zemoso',
  data: [
    { date: 'Mon', value: 30 },
    { date: 'Tue', value: 15 },
    { date: 'Wen', value: 12 },
    { date: 'Thu', value: 40 },
    { date: 'Fri', value: 20 },
    { date: 'Sat', value: 5 },
    { date: 'Sun', value: 30 },
  ],
};

export const trustGroupGraphData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
    ],
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 15 },
      { date: '3rd Dec', value: 12 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 20 },
      { date: '6th Dec', value: 5 },
      { date: '7th Dec', value: 30 },
    ],
  },
];

export const trustScoreGraphData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
      { date: '8th Dec', value: 40 },
      { date: '9th Dec', value: 12 },
      { date: '10th Dec', value: 25 },
      { date: '11th Dec', value: 30 },
      { date: '12th Dec', value: 25 },
      { date: '13th Dec', value: 30 },
      { date: '14th Dec', value: 25 },
      { date: '15th Dec', value: 30 },
      { date: '16th Dec', value: 25 },
      { date: '17th Dec', value: 30 },
    ],
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
      { date: '8th Dec', value: 40 },
      { date: '9th Dec', value: 12 },
      { date: '10th Dec', value: 25 },
      { date: '11th Dec', value: 30 },
      { date: '12th Dec', value: 25 },
      { date: '13th Dec', value: 30 },
      { date: '14th Dec', value: 25 },
      { date: '15th Dec', value: 30 },
      { date: '16th Dec', value: 25 },
      { date: '17th Dec', value: 30 },
    ],
  },
];
export const trustTabData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
    ],
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 15 },
      { date: '3rd Dec', value: 12 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 20 },
      { date: '6th Dec', value: 5 },
      { date: '7th Dec', value: 30 },
    ],
  },
];
export const doNotTrustTabData = [
  {
    value: 'zemoso',
    name: 'zemoso',
    data: [
      { date: '1st Dec', value: 30 },
      { date: '2nd Dec', value: 5 },
      { date: '3rd Dec', value: 20 },
      { date: '4th Dec', value: 40 },
      { date: '5th Dec', value: 12 },
      { date: '6th Dec', value: 25 },
      { date: '7th Dec', value: 30 },
    ],
    color: '#ff3162',
  },
  {
    value: 'stanford',
    name: 'stanford',
    data: [
      { date: 'Mon', value: 30 },
      { date: 'Tue', value: 15 },
      { date: 'Wen', value: 12 },
      { date: 'Thu', value: 40 },
      { date: 'Fri', value: 20 },
      { date: 'Sat', value: 5 },
      { date: 'Sun', value: 30 },
    ],
    color: '#ff3162',
  },
];
export const domains = [
  { name: 'zemosolabs', value: 'zemosolabs' },
  { name: 'stanford', value: 'zemostanfordsolabs' },
];
