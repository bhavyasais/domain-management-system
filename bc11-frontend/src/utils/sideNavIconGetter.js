import Graph from '../assets/atoms_icon_graph.svg';
import Pie from '../assets/atoms_icon_pie.svg';
import Sheet from '../assets/atoms_icon_excel.svg';
import Mail from '../assets/mail.png';
const sideNavIcons = [
  { icon: Pie, active: false },
  { icon: Graph, active: true },
  { icon: Mail, active: true },
  { icon: Sheet, active: false },
];

const getSideBarIcons = () => sideNavIcons;

const sideNavIconGetter = {
  getSideBarIcons,
};

export default sideNavIconGetter;
