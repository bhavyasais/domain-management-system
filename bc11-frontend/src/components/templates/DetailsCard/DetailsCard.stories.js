import React, { useState } from 'react';
import DetailsCard from './DetailsCard';
import { DESCRIPTION, SOURCE_DETAILS_HEADER, THREAT_DETAILS_HEADER } from '../../../utils/Strings';

export default {
  title: 'Templates/DetailsCard',
};

export const detailsCard = () => {
  const [state, setState] = useState({ name: 'Do not trust', value: 'Do not trust' });
  const trustGroups = [
    { name: 'Do not trust', value: 'Do not trust' },
    { name: 'Trusted', value: 'Trusted' },
  ];
  const handleOnChange = (event) => {
    setState({ name: event, value: event });
  };
  var cardDetails = {
    id: '#12345',
    Domain: 'Do not trust',
    From: 'Akshay.sunder@gmail.com',
    threatType: 'Spam',
    trustScore: 22,
    trustGroups: trustGroups,
    onChange: { handleOnChange },
    time: '14-12-2019, 12:43pm',
    threatLevel: 'High',
    description: DESCRIPTION,
    emailDetails: DESCRIPTION,
  };

  return (
    <DetailsCard
      cardDetails={cardDetails}
      sourceDetailsHeader={SOURCE_DETAILS_HEADER}
      threatDetailsHeader={THREAT_DETAILS_HEADER}
      selectedItem={state}
    />
  );
};
