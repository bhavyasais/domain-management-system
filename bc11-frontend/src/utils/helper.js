import { RELATIONSHIP_COLORS, TRUSTGROUP_COLORS } from './Strings';

export const FixDomainRowTrustGroup = (array) =>
  array.map((row, idx) => ({
    ...row,
    trustGroup: {
      backgroundColor: TRUSTGROUP_COLORS[row?.trustGroup],
      label: row?.trustGroup,
      idx,
    },
  }));

export const UpdateTrustGroups = (array) =>
  array.map((value) => ({
    backgroundColor: TRUSTGROUP_COLORS[value.name] ?? '#eeeeee',
    label: value.name,
  }));

export const UpdateDomainsValue = (array, data) => {
  const newData = array.map((row) => {
    return row.id === data.id ? { ...data } : { ...row };
  });
  return newData;
};

export const FixDomainRow = (array) =>
  array.map((row) => ({
    ...row,
    relationship: {
      backgroundColor: RELATIONSHIP_COLORS[row?.relationship],
      label: row?.relationship,
    },
  }));

export const CreateTrustGroupPermissionArray = (array, trustgroups) => {
  const newArray = Array.from({ length: trustgroups.length }, () => ({}));

  trustgroups.forEach((obj, idx) => {
    newArray[idx].trustGroupName = obj.groupName;
    newArray[idx].trust_groups_id = obj.id;
    newArray[idx].viewFile = false;
    newArray[idx].commentFile = false;
    newArray[idx].editFile = false;
    newArray[idx].transferFile = false;
  });

  const finalArray = newArray.map((obj) => {
    const newObj = obj;
    const matchedObjects = array.filter((item) => item.trust_group_name === obj.trustGroupName);
    matchedObjects.forEach((item) => {
      if (item.permission_type === 'READ') {
        newObj.viewFile = item.is_active;
        newObj.viewFileId = item.id;
        newObj.viewFileAccessId = item.access_permission_types_id;
      } else if (item.permission_type === 'COMMENT') {
        newObj.commentFile = item.is_active;
        newObj.commentFileId = item.id;
        newObj.commentFileAccessId = item.access_permission_types_id;
      } else if (item.permission_type === 'EDIT') {
        newObj.editFile = item.is_active;
        newObj.editFileId = item.id;
        newObj.editFileAccessId = item.access_permission_types_id;
      } else if (item.permission_type === 'TRANSFER OWNERSHIP') {
        newObj.transferFile = item.is_active;
        newObj.transferFileId = item.id;
        newObj.transferFileAccessId = item.access_permission_types_id;
      }
    });
    return newObj;
  });
  return finalArray;
};

export const createUpdateTrustGroupParam = (item, colId, checked) => {
  const newObj = {};
  newObj.is_active = checked;
  newObj.trust_groups_id = item.trust_groups_id;

  if (colId === 'viewFile') {
    newObj.id = item.viewFileId;
    newObj.access_permission_types_id = item.viewFileAccessId;
  } else if (colId === 'commentFile') {
    newObj.id = item.commentFileId;
    newObj.access_permission_types_id = item.commentFileAccessId;
  } else if (colId === 'editFile') {
    newObj.id = item.editFileId;
    newObj.access_permission_types_id = item.editFileAccessId;
  } else if (colId === 'transferFile') {
    newObj.id = item.transferFileId;
    newObj.access_permission_types_id = item.transferFileAccessId;
  }
  return newObj;
};

export const getPermission = (array) => {
  array.forEach((item) => {
    if (item?.transferFile) {
      localStorage.setItem(item.trustGroupName, 'writer');
    } else if (item?.editFile) {
      localStorage.setItem(item.trustGroupName, 'writer');
    } else if (item?.commentFile) {
      localStorage.setItem(item.trustGroupName, 'commenter');
    } else if (item?.viewFile) {
      localStorage.setItem(item.trustGroupName, 'reader');
    } else {
      localStorage.setItem(item.trustGroupName, 'none');
    }
  });
};

export const createNewDomainName=(list) =>{
  let domain_name = 'New Domain';
  while (list?.find((item) => item?.domain_name === domain_name)) {
    domain_name = domain_name.concat(" ",list.length);
  }
  return domain_name;
}

export const validateDomainName=(list, node, gridApi, gridColumnApi) =>{
  if (node?.column?.colId === 'domain_name') {
    if (list?.find((item) => item?.domain_name?.trim() === node?.data?.domain_name?.trim())) {
      gridApi.startEditingCell({
        rowIndex: node?.rowIndex,
        colKey:gridColumnApi?.getAllColumns()?.[1]?.colId,
      });
    }
  }
}

export const getDomainCount=(list, node)=> {
  var count = 0;
  for (var i = 0; i < list.length; i++) {
    if (list?.[i]?.domain_name.trim() === node?.oldValue.trim()){ count++}
  }
  return count;
}
