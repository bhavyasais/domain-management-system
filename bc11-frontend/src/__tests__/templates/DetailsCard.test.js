import React from 'react';
import renderer from 'react-test-renderer';
import { DESCRIPTION, SOURCE_DETAILS_HEADER, THREAT_DETAILS_HEADER } from '../../utils/Strings';
import DetailsCard from '../../components/templates/DetailsCard/DetailsCard';
const handleOnChange = jest.fn();
const trustGroups = [
  { name: 'Do not trust', value: 'Do not trust' },
  { name: 'Trusted', value: 'Trusted' },
];

var cardDetails = {
  id: '#12345',
  Domain: 'Do not trust',
  From: 'Akshay.sunder@gmail.com',
  threatType: 'Spam',
  trustScore: 22,
  trustGroups: trustGroups,
  onChange: { handleOnChange },
  time: '14-12-2019, 12:43pm',
  threatLevel: 'High',
  description: DESCRIPTION,
  emailDetails: DESCRIPTION,
};

describe('<DetailsCard/>', () => {
  it('Renders correctly', () => {
    let wrapper;
    wrapper = renderer.create(
      <DetailsCard
        cardDetails={cardDetails}
        sourceDetailsHeader={SOURCE_DETAILS_HEADER}
        threatDetailsHeader={THREAT_DETAILS_HEADER}
        selectedItem={{ name: 'Do not trust', value: 'Do not trust' }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
