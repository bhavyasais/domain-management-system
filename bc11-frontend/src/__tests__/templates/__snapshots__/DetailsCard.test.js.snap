// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsCard/> Renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
  style={
    Object {
      "borderColor": undefined,
    }
  }
>
  <div
    className="MuiCardHeader-root makeStyles-threat-2"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h4 MuiTypography-displayBlock"
      >
        Threat Details
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <div>
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div>
              <span
                aria-hidden={true}
                className="material-icons MuiIcon-root"
                style={
                  Object {
                    "color": undefined,
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </div>
          </span>
        </button>
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle MuiDivider-light"
  />
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root makeStyles-margin-5 makeStyles-subject-4 MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
        >
          #12345
          : Subject
        </h6>
        <div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "color": undefined,
              }
            }
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label makeStyles-buttonLabel-10"
            >
              <span
                className="MuiTypography-root makeStyles-buttonLabel-10 MuiTypography-caption"
              >
                View Source
              </span>
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-details-7 MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            Domain
          </span>
          <p
            className="MuiTypography-root makeStyles-contentStyle-6 MuiTypography-body1"
          >
            Do not trust
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            From
          </span>
          <p
            className="MuiTypography-root makeStyles-contentStyle-6 MuiTypography-body1"
          >
            Akshay.sunder@gmail.com
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            Description
          </span>
          <p
            className="MuiTypography-root makeStyles-contentStyle-6 MuiTypography-body1"
            style={
              Object {
                "width": "378px",
              }
            }
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            Treat type
          </span>
          <p
            className="MuiTypography-root makeStyles-contentStyle-6 MuiTypography-body1"
          >
            Spam
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            Treat level
          </span>
          <p
            className="MuiTypography-root makeStyles-contentStyle-6 MuiTypography-body1"
          >
            High
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            Trust score
          </span>
          <p
            className="MuiTypography-root makeStyles-contentStyle-6 MuiTypography-body1"
          >
            22
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            Source date time
          </span>
          <p
            className="MuiTypography-root makeStyles-contentStyle-6 MuiTypography-body1"
          >
            14-12-2019, 12:43pm
          </p>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item"
        >
          <span
            className="MuiTypography-root makeStyles-margin-5 MuiTypography-button"
          >
            Group
          </span>
          <div
            className={null}
          >
            <div
              className="MuiFormControl-root Dropdown-formControl-11 makeStyles-dropDown-9"
            >
              <div
                className="MuiInputBase-root MuiOutlinedInput-root Dropdown-dropdown-13 MuiInputBase-formControl"
                onClick={[Function]}
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="simple-select-label simple-select"
                  className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="simple-select"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  Do not trust
                </div>
                <input
                  aria-hidden={true}
                  className="MuiSelect-nativeInput"
                  onAnimationStart={[Function]}
                  onChange={[Function]}
                  required={false}
                  tabIndex={-1}
                  value="Do not trust"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden={true}
                  className="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
                  style={
                    Object {
                      "paddingLeft": 8,
                    }
                  }
                >
                  <legend
                    className="PrivateNotchedOutline-legend-15"
                    style={
                      Object {
                        "width": 0.01,
                      }
                    }
                  >
                    <span
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "&#8203;",
                        }
                      }
                    />
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
